servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -630,7 +630,9 @@ void markHealthy() {
             cancelIfHealthCheck(oldState);
         }
 
-        void markUnhealthy(ConnectionFactory<Addr, ? extends C> connectionFactory) {
+        void markUnhealthy(final Throwable cause, final ConnectionFactory<Addr, ? extends C> connectionFactory) {
+            LOGGER.debug("Load balancer for {}: failed to open a new connection to the host on address {}.",
+                    targetResource, address, cause);
```

Comment:
I planned to include the cause in pre-existing log message at line 649, but then decided to log it for all cases (even if the host became non-active at the time we tried to open a new connection) for better visibility.

Reply:
It feels the log is duplicated now. Can you please adjust the other log to be more about the number of times failure happened and skip the `failed to open a new connection to the host on address {}`?

E.g.
`"Load balancer for {}: failure #{} ({} consecutive failures will trigger health check)."`

Reply:
Agreed, but we always have to include the host address in the log. Otherwise, we won't be able to track errors per IP. I made a separate log statement for each case inside `for` loop with consistent prefix to allow easier search of the logs related to the same LB and address.